src/PrestaShopBundle/Resources/config/services/core/form/choice_provider.yml

```diff
@@ -158,7 +158,7 @@ services:
   prestashop.core.form.choice_provider.currency_name_by_iso_code:
     class: 'PrestaShop\PrestaShop\Core\Form\ChoiceProvider\CurrencyNameByIsoCodeChoiceProvider'
     arguments:
-      - '@=service("prestashop.core.cldr.repository").getAllCurrencies()'
+      - '@=service("prestashop.core.localization.cldr.locale_repository").getLocale(service("prestashop.adapter.legacy.context").getContext().language.getLocale()).getAllCurrencies()'
```

Comment:
wow!! 🤪
I get the logic but it's quite a long expression, do you think there might a service that could do this more easily? Although I don't have the solution, it's a question? ^^

Reply:
Well, I also thought about a dedicated service, but I wonder if it's a good idea.
As we rely on a context attribute (language), I do not want to see a service initialized too early and see it empty.

Reply:
Yes I see the problem, maybe we would need a service with the legacy context, which would have `getContextLocale` `getContextCurrency` (rough idea..) which would be resolved on the fly
But anyway it's not the purpose of this PR